### PR TITLE
Core implementation

### PR DESCRIPTION
**Pull request by mykhailonakko**
Created a framework structure:
- added necessary dependencies.
- added Report Portal listener.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,39 @@
+target/
+!.mvn/wrapper/maven-wrapper.jar
+!**/src/main/**/target/
+!**/src/test/**/target/
```
Comment (NovoDan) [inline .gitignore]: just one 'target/' would be enough

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+### IntelliJ IDEA ###
+.idea
```
Comment (NovoDan) [inline .gitignore]: same here

File: src/main/java/org/rptest/core/logger/JavaUtilLogger.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    public void error(String message) {
+        logger.log(Level.SEVERE, message);
+    }
+
+    private String getCallerClassName() {
+        StackTraceElement[] stElements = Thread.currentThread().getStackTrace();
+        String className = stElements[3].getClassName();
+        try {
+            return Class.forName(className).getName();
```
Comment (NovoDan) [inline src/main/java/org/rptest/core/logger/JavaUtilLogger.java:26]: Doesn't it will return same value as stored in 'className'?

**Review by NovoDan — CHANGES_REQUESTED** [review]

**Comment by ihor-dynka** [comment]
in pom.xml all dependencies and plugins version move to properties section. It increases readability and adds option to be able to manage version via command line

File: .gitignore (hunk 1 above)
Comment (mykhailonakko) [reply .gitignore]: Thanks, will fix that.

File: src/main/java/org/rptest/core/logger/JavaUtilLogger.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        logger.log(Level.INFO, message);
+    }
+
+    public void error(String message) {
+        logger.log(Level.SEVERE, message);
+    }
+
+    private String getCallerClassName() {
+        StackTraceElement[] stElements = Thread.currentThread().getStackTrace();
+        String className = stElements[3].getClassName();
```
Comment (ihor-dynka) [inline src/main/java/org/rptest/core/logger/JavaUtilLogger.java]: please avoid magic numbers

File: src/main/java/org/rptest/core/browser/PwBrowser.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import com.microsoft.playwright.BrowserContext;
+import com.microsoft.playwright.Page;
+import com.microsoft.playwright.Playwright;
+
+import java.util.Objects;
+
+import static com.microsoft.playwright.BrowserType.LaunchOptions;
+import static org.rptest.core.config.Property.COMMON_PROPERTY;
+
+public class PwBrowser {
```
Comment (ihor-dynka) [inline src/main/java/org/rptest/core/browser/PwBrowser.java]: avoid abbreviation, use full names

File: src/main/java/org/rptest/core/elements/UiElementFactory.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import org.rptest.core.browser.PwBrowser;
+import org.rptest.core.config.Property;
+import org.rptest.core.constants.UiLibrary;
+
+@UtilityClass
+public class UiElementFactory {
+    private static final UiLibrary library = Enum.valueOf(UiLibrary.class, Property.COMMON_PROPERTY.uiLibrary().toUpperCase());
+
+    public static UiElement findElementById(String id) {
+        switch (library) {
```
Comment (ihor-dynka) [inline src/main/java/org/rptest/core/elements/UiElementFactory.java]: this login should be executed once

File: src/main/java/org/rptest/core/utilities/FileUtils.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.List;
+
+@UtilityClass
+public class FileUtils {
+    public static List<CSVRecord> readCSV(String filePath) throws IOException {
+        Reader reader = new FileReader(filePath);
```
Comment (ihor-dynka) [inline src/main/java/org/rptest/core/utilities/FileUtils.java:16]: another option can be using try with resources and rethrow another exception

**Comment by ihor-dynka** [comment]
- please add option to run the tests in parallel mode (investigate the difference between class and methods parallelization, forks vs threads)

**Comment by ihor-dynka** [comment]
- add branch policies to allow merging only after at least 1 approve and all comments resolved

File: src/main/java/org/rptest/core/logger/JavaUtilLogger.java (hunk 3 above)
Comment (mykhailonakko) [reply src/main/java/org/rptest/core/logger/JavaUtilLogger.java:26]: Good question. 

I tried with a few classes and it returned the same thing, but here's what chatGPT has to say about it:

_You're correct that in most cases, `stElements[3].getClassName();` and `Class.forName(className).getName();` will return the same value. The main reason for using `Class.forName(className)` in the try-catch block is to ensure that the class name obtained from the stack trace actually exists and can be loaded by the Java ClassLoader.

In some cases, the class might not be available at runtime due to various reasons such as class loading issues, classpath configuration problems, or the class being removed or relocated in the project. By using `Class.forName(className)`, the code checks if the class can be loaded successfully. If a `ClassNotFoundException` is thrown, it indicates that the class could not be found or loaded, and an error message is logged accordingly.

However, if you are confident that the class will always be available at runtime, you can simplify the method by removing the try-catch block and returning the class name directly from the stack trace element_

```java
private String getCallerClassName() {
    StackTraceElement[] stElements = Thread.currentThread().getStackTrace();
    return stElements[3].getClassName();
}
```

_This simplified version of the method assumes that the class name obtained from the stack trace will always be valid and loadable. Keep in mind that if there's any issue with loading the class, this simplified version will not log an error message and will not handle the potential `ClassNotFoundException`._

File: .gitignore (hunk 1 above)
Comment (mykhailonakko) [reply .gitignore]: fixed

**Review by ihor-dynka — APPROVED** [review]

**Pull request merged**